pkg/storaged/dialog.js

```diff
@@ -287,12 +287,18 @@
             $(this).parents(".input-group").removeClass("open");
         });
 
+        var set_choices_counter = 0;
+
         function combobox_set_choices(name, choices) {
             if (typeof choices == 'function') {
-                $.when(choices(get_field_values())).then(function (result) {
-                    if (result !== false)
-                        combobox_set_choices(name, result);
-                });
+                set_choices_counter += 1;
+                var count = set_choices_counter;
+                choices(get_field_values(),
+                        function (ch) {
+                            console.log("R", ch, count, set_choices_counter);
```

Comment:
Debugging leftover?

Admittedly I don't understand this. `count` is defined to be `set_choices_counter`, and never gets changed. So how would `if (count == set_choices_counter)` ever be false? What is this counter meant to achieve?

Reply:
> Debugging leftover?

Indeed!

> count is defined to be set_choices_counter, and never gets changed.

It is `set_choices_counter` that might get changed.  Every new call to `combobox_set_choices` should cancel all the background activity of all previous calls.  But that's hard, so we just ignore callback invocations that come too late.